docs/index.adoc

```diff
@@ -11401,6 +11404,31 @@ Multi-line text blocks can be used in command and option descriptions, headers a
 
 For more details, see https://www.infoq.com/articles/java-text-blocks/[this article] by Java Language Architect Brian Goetz.
 
+=== Usage of `System.exit`
+
```

Comment:
Instead of adding a new subsection **Usage of `System.exit`** to section **9. Executing Commands**, I propose we put all the text below in a new subsection under [21. Tips & Tricks](https://picocli.info/#_tips_tricks).

Then, we link to that new subsection from the [9.1. Exit Code](https://picocli.info/#_exit_code) subsection, by adding a sentence like this to the end of the 9.1. Exit Code subsection:

> There are cases where calling `System.exit` is not ideal. See < < NEW SECTION > > for details.